### DISCUSSION TITLE
Problem on resume playback

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1432] [Bug Reports]
RPI3B+, Updated to: 20230918, Infineon Merus Amp DAC, Local display via RPi Official 7in touchscreen (using Features / Others / Browser on RPi)

I use this system to mostly play web radio stations during the day and it can be in use for many hours each day and for the most part it is fine. However if I need to go out I stop rAudio via the Stop button on the top bar or via the central logo touch menu.  I do not turn the RPi itself off.  When I get back I simply press the Play button to resume the original web radio station,  In most cases, but not all, the radio will start to play, no problem.  On the occasions when no output is heard I can see from the display that the station appears to be playing (the 3 dots indicator is 'moving' and the rest of the system pages are accessible) but there is no audio output. It then takes a reboot of the RPi in order to restart the audio output and (usually) the system then plays normally . I don't know if this problem happens with albums or other music sources as this is not this systems general use case but I suspect it will. 

From the output of a "journalctl -xe" command (before a reboot) I can see a couple of problems which might be the cause:

-- A start job for unit alsa-restore.service has begun execution.
-- 
-- The job identifier is 224.
Sep 21 15:32:45 RAConservatory systemd[1]: Manage Sound Card State (restore and store) was skipped because of an unmet condition check (ConditionPathExists=/etc/alsa/state-daemon.conf).
-- Subject: A start job for unit alsa-state.service has finished successfully
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- A start job for unit alsa-state.service has finished successfully.
-- 
-- The job identifier is 300.
Sep 21 15:32:45 RAConservatory avahi-daemon[320]: Found user 'avahi' (UID 969) and group 'avahi' (GID 969).
Sep 21 15:32:45 RAConservatory avahi-daemon[320]: Successfully dropped root privileges.
Sep 21 15:32:45 RAConservatory avahi-daemon[320]: avahi-daemon 0.8 starting up.
Sep 21 15:32:45 RAConservatory alsactl[335]: alsa-lib main.c:1560:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -2
Sep 21 15:32:45 RAConservatory systemd[1]: Finished Save/Restore Sound Card State.

and 

Sep 21 15:33:20 RAConservatory systemd[1]: Started MPD idle.
-- Subject: A start job for unit mpdidle.service has finished successfully
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- A start job for unit mpdidle.service has finished successfully.
-- 
-- The job identifier is 552.
Sep 21 15:33:22 RAConservatory startup.sh[961]: Traceback (most recent call last):
Sep 21 15:33:22 RAConservatory startup.sh[961]:   File "/srv/http/bash/websocket-push.py", line 8, in <module>
Sep 21 15:33:22 RAConservatory startup.sh[961]:     ws = create_connection( 'ws://'+ ip +':8080' )
Sep 21 15:33:22 RAConservatory startup.sh[961]:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:22 RAConservatory startup.sh[961]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 608, in create_connection
Sep 21 15:33:22 RAConservatory startup.sh[961]:     websock.connect(url, **options)
Sep 21 15:33:22 RAConservatory startup.sh[961]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 249, in connect
Sep 21 15:33:22 RAConservatory startup.sh[961]:     self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
Sep 21 15:33:22 RAConservatory startup.sh[961]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:22 RAConservatory startup.sh[961]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 129, in connect
Sep 21 15:33:22 RAConservatory startup.sh[961]:     sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
Sep 21 15:33:22 RAConservatory startup.sh[961]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:22 RAConservatory startup.sh[961]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 204, in _open_socket
Sep 21 15:33:22 RAConservatory startup.sh[961]:     raise err
Sep 21 15:33:22 RAConservatory startup.sh[961]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 184, in _open_socket
Sep 21 15:33:22 RAConservatory startup.sh[961]:     sock.connect(address)
Sep 21 15:33:22 RAConservatory startup.sh[961]: ConnectionRefusedError: [Errno 111] Connection refused
Sep 21 15:33:23 RAConservatory startup.sh[968]: Traceback (most recent call last):
Sep 21 15:33:23 RAConservatory startup.sh[968]:   File "/srv/http/bash/websocket-push.py", line 8, in <module>
Sep 21 15:33:23 RAConservatory startup.sh[968]:     ws = create_connection( 'ws://'+ ip +':8080' )
Sep 21 15:33:23 RAConservatory startup.sh[968]:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:23 RAConservatory startup.sh[968]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 608, in create_connection
Sep 21 15:33:23 RAConservatory startup.sh[968]:     websock.connect(url, **options)
Sep 21 15:33:23 RAConservatory startup.sh[968]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 249, in connect
Sep 21 15:33:23 RAConservatory startup.sh[968]:     self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
Sep 21 15:33:23 RAConservatory startup.sh[968]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:23 RAConservatory startup.sh[968]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 129, in connect
Sep 21 15:33:23 RAConservatory startup.sh[968]:     sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
Sep 21 15:33:23 RAConservatory startup.sh[968]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:23 RAConservatory startup.sh[968]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 204, in _open_socket
Sep 21 15:33:23 RAConservatory startup.sh[968]:     raise err
Sep 21 15:33:23 RAConservatory startup.sh[968]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 184, in _open_socket
Sep 21 15:33:23 RAConservatory startup.sh[968]:     sock.connect(address)
Sep 21 15:33:23 RAConservatory startup.sh[968]: ConnectionRefusedError: [Errno 111] Connection refused
Sep 21 15:33:24 RAConservatory startup.sh[975]: Traceback (most recent call last):
Sep 21 15:33:24 RAConservatory startup.sh[975]:   File "/srv/http/bash/websocket-push.py", line 8, in <module>
Sep 21 15:33:24 RAConservatory startup.sh[975]:     ws = create_connection( 'ws://'+ ip +':8080' )
Sep 21 15:33:24 RAConservatory startup.sh[975]:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:24 RAConservatory startup.sh[975]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 608, in create_connection
Sep 21 15:33:24 RAConservatory startup.sh[975]:     websock.connect(url, **options)
Sep 21 15:33:24 RAConservatory startup.sh[975]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 249, in connect
Sep 21 15:33:24 RAConservatory startup.sh[975]:     self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
Sep 21 15:33:24 RAConservatory startup.sh[975]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:24 RAConservatory startup.sh[975]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 129, in connect
Sep 21 15:33:24 RAConservatory startup.sh[975]:     sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
Sep 21 15:33:24 RAConservatory startup.sh[975]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:24 RAConservatory startup.sh[975]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 204, in _open_socket
Sep 21 15:33:24 RAConservatory startup.sh[975]:     raise err
Sep 21 15:33:24 RAConservatory startup.sh[975]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 184, in _open_socket
Sep 21 15:33:24 RAConservatory startup.sh[975]:     sock.connect(address)
Sep 21 15:33:24 RAConservatory startup.sh[975]: ConnectionRefusedError: [Errno 111] Connection refused
Sep 21 15:33:24 RAConservatory startup.sh[981]: Traceback (most recent call last):
Sep 21 15:33:24 RAConservatory startup.sh[981]:   File "/srv/http/bash/websocket-push.py", line 8, in <module>
Sep 21 15:33:24 RAConservatory startup.sh[981]:     ws = create_connection( 'ws://'+ ip +':8080' )
Sep 21 15:33:24 RAConservatory startup.sh[981]:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:24 RAConservatory startup.sh[981]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 608, in create_connection
Sep 21 15:33:24 RAConservatory startup.sh[981]:     websock.connect(url, **options)
Sep 21 15:33:24 RAConservatory startup.sh[981]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 249, in connect
Sep 21 15:33:24 RAConservatory startup.sh[981]:     self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
Sep 21 15:33:24 RAConservatory startup.sh[981]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:24 RAConservatory startup.sh[981]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 129, in connect
Sep 21 15:33:24 RAConservatory startup.sh[981]:     sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
Sep 21 15:33:24 RAConservatory startup.sh[981]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:24 RAConservatory startup.sh[981]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 204, in _open_socket
Sep 21 15:33:24 RAConservatory startup.sh[981]:     raise err
Sep 21 15:33:24 RAConservatory startup.sh[981]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 184, in _open_socket
Sep 21 15:33:24 RAConservatory startup.sh[981]:     sock.connect(address)
Sep 21 15:33:24 RAConservatory startup.sh[981]: ConnectionRefusedError: [Errno 111] Connection refused
Sep 21 15:33:26 RAConservatory startup.sh[1114]: Traceback (most recent call last):
Sep 21 15:33:26 RAConservatory startup.sh[1114]:   File "/srv/http/bash/websocket-push.py", line 8, in <module>
Sep 21 15:33:26 RAConservatory startup.sh[1114]:     ws = create_connection( 'ws://'+ ip +':8080' )
Sep 21 15:33:26 RAConservatory startup.sh[1114]:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:26 RAConservatory startup.sh[1114]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 608, in create_connection
Sep 21 15:33:26 RAConservatory startup.sh[1114]:     websock.connect(url, **options)
Sep 21 15:33:26 RAConservatory startup.sh[1114]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 249, in connect
Sep 21 15:33:26 RAConservatory startup.sh[1114]:     self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
Sep 21 15:33:26 RAConservatory startup.sh[1114]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:26 RAConservatory startup.sh[1114]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 129, in connect
Sep 21 15:33:26 RAConservatory startup.sh[1114]:     sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
Sep 21 15:33:26 RAConservatory startup.sh[1114]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:26 RAConservatory startup.sh[1114]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 204, in _open_socket
Sep 21 15:33:26 RAConservatory startup.sh[1114]:     raise err
Sep 21 15:33:26 RAConservatory startup.sh[1114]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 184, in _open_socket
Sep 21 15:33:26 RAConservatory startup.sh[1114]:     sock.connect(address)
Sep 21 15:33:26 RAConservatory startup.sh[1114]: ConnectionRefusedError: [Errno 111] Connection refused
Sep 21 15:33:27 RAConservatory startup.sh[1147]: Traceback (most recent call last):
Sep 21 15:33:27 RAConservatory startup.sh[1147]:   File "/srv/http/bash/websocket-push.py", line 8, in <module>
Sep 21 15:33:27 RAConservatory startup.sh[1147]:     ws = create_connection( 'ws://'+ ip +':8080' )
Sep 21 15:33:27 RAConservatory startup.sh[1147]:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:27 RAConservatory startup.sh[1147]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 608, in create_connection
Sep 21 15:33:27 RAConservatory startup.sh[1147]:     websock.connect(url, **options)
Sep 21 15:33:27 RAConservatory startup.sh[1147]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 249, in connect
Sep 21 15:33:27 RAConservatory startup.sh[1147]:     self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
Sep 21 15:33:27 RAConservatory startup.sh[1147]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:27 RAConservatory startup.sh[1147]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 129, in connect
Sep 21 15:33:27 RAConservatory startup.sh[1147]:     sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
Sep 21 15:33:27 RAConservatory startup.sh[1147]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:27 RAConservatory startup.sh[1147]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 204, in _open_socket
Sep 21 15:33:27 RAConservatory startup.sh[1147]:     raise err
Sep 21 15:33:27 RAConservatory startup.sh[1147]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 184, in _open_socket
Sep 21 15:33:27 RAConservatory startup.sh[1147]:     sock.connect(address)
Sep 21 15:33:27 RAConservatory startup.sh[1147]: ConnectionRefusedError: [Errno 111] Connection refused
Sep 21 15:33:28 RAConservatory startup.sh[1193]: Traceback (most recent call last):
Sep 21 15:33:28 RAConservatory startup.sh[1193]:   File "/srv/http/bash/websocket-push.py", line 8, in <module>
Sep 21 15:33:28 RAConservatory startup.sh[1193]:     ws = create_connection( 'ws://'+ ip +':8080' )
Sep 21 15:33:28 RAConservatory startup.sh[1193]:          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:28 RAConservatory startup.sh[1193]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 608, in create_connection
Sep 21 15:33:28 RAConservatory startup.sh[1193]:     websock.connect(url, **options)
Sep 21 15:33:28 RAConservatory startup.sh[1193]:   File "/usr/lib/python3.11/site-packages/websocket/_core.py", line 249, in connect
Sep 21 15:33:28 RAConservatory startup.sh[1193]:     self.sock, addrs = connect(url, self.sock_opt, proxy_info(**options),
Sep 21 15:33:28 RAConservatory startup.sh[1193]:                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:28 RAConservatory startup.sh[1193]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 129, in connect
Sep 21 15:33:28 RAConservatory startup.sh[1193]:     sock = _open_socket(addrinfo_list, options.sockopt, options.timeout)
Sep 21 15:33:28 RAConservatory startup.sh[1193]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Sep 21 15:33:28 RAConservatory startup.sh[1193]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 204, in _open_socket
Sep 21 15:33:28 RAConservatory startup.sh[1193]:     raise err
Sep 21 15:33:28 RAConservatory startup.sh[1193]:   File "/usr/lib/python3.11/site-packages/websocket/_http.py", line 184, in _open_socket
Sep 21 15:33:28 RAConservatory startup.sh[1193]:     sock.connect(address)
Sep 21 15:33:28 RAConservatory startup.sh[1193]: ConnectionRefusedError: [Errno 111] Connection refused
Sep 21 15:33:28 RAConservatory systemd[1]: startup.service: Deactivated successfully.

This problem seems to have started recently as playback on this system used to be flawless but I can't be sure just when or with which version.  The system hardware hasn't changed in several years and as it is used a lot by my wife, it is getting harder and harder to explain why the radio isn't playing - particularly when I don't know the actual cause :-)

Chris

**Comment by ceejayemm** [comment]
@rern 

It seems like this issue is related to using a local display via the 'Browser on RPi'  (via the local Firefox implementation) feature. If I control this system via a web browser on another PC or via the Android app on a mobile device then this issue is not seen.

Chris

**Comment by ceejayemm** [comment]
@rern 
These errors are still present after updating to 20231020

Regards.

Chris

**Reply by rern** [reply]
May be https://github.com/rern/rAudio/discussions/1435#discussioncomment-7345671 ?

**Reply by ceejayemm** [reply]
@rern 
I had a look at #1435, I had most recently rebuilt my system using i20230828 (or may be i20230918 - I can't remember which) and updated first to 202310131 and then to 20231020 still with the issues noted above.  This gave me kernel 6.1.47-1, so lower than the versions noted in #1435.  I don't normally use pacman but in this case I will give it a go and see if it makes any difference.

Thanks

Chris

**Comment by ceejayemm** [comment]
@rern 
I ahve applied  'pacman -Syu' as suggested in #1435, it seems to have given me kernel 6.1.58-2 plus a whole load of other updates.  The errors noted above seem to have now disappeared  when navigating via the touchscreen - however there a  bunch of others which still appear (from journalctl -f whilst navigating via touchscreen):

Oct 21 14:54:54 RAConservatory sudo[3094]: pam_unix(sudo:session): session closed for user root
Oct 21 14:54:57 RAConservatory nginx[447]: 2023/10/21 14:54:57 [error] 447#447: *18 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194" while reading response header from upstream, client: 127.0.0.1, server: localhost, request: "GET /settings.php?p=player HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "localhost", referrer: "http://localhost/settings.php?p=networks"
Oct 21 14:54:57 RAConservatory nginx[447]: 2023/10/21 14:54:57 [error] 447#447: *18 FastCGI sent in stderr: "; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194" while reading upstream, client: 127.0.0.1, server: localhost, request: "GET /settings.php?p=player HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "localhost", referrer: "http://localhost/settings.php?p=networks"
Oct 21 14:54:57 RAConservatory nginx[447]: 2023/10/21 14:54:57 [error] 447#447: *18 FastCGI sent in stderr: "; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194; PHP message: PHP Warning:  Undefined array key "icon" in /srv/http/settings.php on line 194" while reading upstream, client: 127.0.0.1, server: localhost, request: "GET /settings.php?p=player HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "localhost", referrer: "http://localhost/settings.php?p=networks"
Oct 21 14:54:58 RAConservatory sudo[3158]:     http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/settings/player-data.sh

Not sure what they relate to as I can't see anything 'missing' or 'faulty' from the user interface.

Please let me know if I can provide any further information to help resolve these issues.

Thanks

Chris
